### ISSUE TITLE
Wrong warning of -Wgnu-variable-sized-type-not-at-end

### ISSUE THREAD

**Issue by Snape3058** [llvm/llvm-project#117559]
```c
struct Wrappee {int b;int a[];};
struct A {                                                                        
  struct B {                                                                      
    struct Wrappee data;  // <--- Correct warning here.
    int plug;                                                                   
  } x;  // <--- Incorrect warning here.
  int plug;                                                                     
};                                                                              
```

The warning on field `x` is incorrect, as the `plug` in B makes it not a flexible array.

Cross-checked with GCC: https://godbolt.org/z/WWcvdG5Wd
